pkg/handler/handler.go

```diff
@@ -50,77 +51,79 @@ func computeETag(data []byte) string {
 // the ability to safely change the spec while serving it.
 type OpenAPIService struct {
 	// rwMutex protects All members of this service.
-	rwMutex sync.RWMutex
+	mutex sync.Mutex
 
 	lastModified time.Time
 
-	jsonCache  handler.HandlerCache
-	protoCache handler.HandlerCache
-	etagCache  handler.HandlerCache
+	specCache  cached.Replaceable[*spec.Swagger]
+	jsonCache  cached.Data[[]byte]
+	protoCache cached.Data[[]byte]
 }
 
 // NewOpenAPIService builds an OpenAPIService starting with the given spec.
-func NewOpenAPIService(spec *spec.Swagger) (*OpenAPIService, error) {
+func NewOpenAPIService(swagger *spec.Swagger) (*OpenAPIService, error) {
+	return NewOpenAPIServiceLazy(cached.NewResultOK(swagger, uuid.New().String()))
+}
+
+// NewOpenAPIServiceLazy builds an OpenAPIService from lazy spec.
+func NewOpenAPIServiceLazy(swagger cached.Data[*spec.Swagger]) (*OpenAPIService, error) {
```

Comment:
Can we ever return an error here?

Reply:
Good catch, we can't, thanks.

Reply:
It means the one above can't return an error either.

Reply:
Fixed, thanks